### PR TITLE
Secrets endpoints and encryption

### LINKED ISSUE #50: SecretController.java:41-43: create an endpoint to update...
The puzzle `44-57fcb942` from #44 has to be resolved: 

https://github.com/tracehubpm/pmo/blob/786aafb3cb357f34ca2c122e20453944c258deb9/src/main/java/git/tracehub/pmo/controller/SecretController.java#L41-L43

The puzzle was created by @hizmailovich on 20-Feb-24. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### LINKED ISSUE #51: DefaultSecrets.java:34-36: encrypt the secret value...
The puzzle `44-48bf6f8b` from #44 has to be resolved: 

https://github.com/tracehubpm/pmo/blob/786aafb3cb357f34ca2c122e20453944c258deb9/src/main/java/git/tracehub/pmo/secret/DefaultSecrets.java#L34-L36

The puzzle was created by @hizmailovich on 20-Feb-24. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by dukris**
closes #51 
closes #50

---

## PR-Codex overview
This PR focuses on adding encryption functionality using Jasypt, updating SQL queries, and improving exception handling in GitHub platform tests.

### Detailed summary
- Added Jasypt encryption for secrets
- Updated SQL queries for better security
- Improved exception handling in GitHub platform tests

> The following files were skipped due to too many changes: `src/main/java/git/tracehub/pmo/secret/Keys.java`, `src/main/java/git/tracehub/pmo/secret/Key.java`, `src/test/java/it/database/DefaultSecretsIT.java`, `src/main/java/git/tracehub/pmo/secret/KeyOf.java`, `README.md`, `src/main/java/git/tracehub/pmo/secret/DefaultSecrets.java`, `src/test/java/git/tracehub/pmo/controller/TicketControllerTest.java`, `src/main/java/git/tracehub/pmo/secret/DefaultKeys.java`, `src/test/java/git/tracehub/pmo/controller/ProjectControllerTest.java`, `src/main/java/git/tracehub/pmo/secret/UniqueSecrets.java`, `src/main/java/git/tracehub/pmo/controller/SecretController.java`, `src/test/java/git/tracehub/pmo/secret/KeyOfTest.java`, `src/main/java/git/tracehub/pmo/secret/EncryptedSecrets.java`, `src/test/java/git/tracehub/pmo/extensions/MkArgumentException.java`, `src/test/java/it/database/DefaultKeysIT.java`, `src/test/java/it/web/RetrieveSecretsByProjectITCase.java`, `src/test/java/git/tracehub/pmo/secret/DefaultSecretsTest.java`, `src/test/java/git/tracehub/pmo/controller/AdviceControllerTest.java`, `src/test/java/git/tracehub/pmo/controller/SecretControllerTest.java`, `src/test/java/git/tracehub/pmo/secret/UniqueSecretsTest.java`, `src/test/java/git/tracehub/pmo/secret/EncryptedSecretsTest.java`, `src/test/java/git/tracehub/pmo/secret/DefaultKeysTest.java`, `src/test/java/it/web/UpdateSecretITCase.java`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/test/java/git/tracehub/pmo/controller/AdviceControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ */
+final class AdviceControllerTest {
+
+    @Test
+    void handlesMethodArgumentNotValidException() {
+        final String message = "MethodArgumentNotValidException";
+        final MethodArgumentNotValidException exception =
+            Mockito.mock(MethodArgumentNotValidException.class);
+        final BindingResult result = Mockito.mock(BindingResult.class);
+        Mockito.when(exception.getBindingResult()).thenReturn(result);
```
Comment (h1alexbel) [inline src/test/java/git/tracehub/pmo/controller/AdviceControllerTest.java:47]: prefer [JUnit extensions](https://junit.org/junit5/docs/current/user-guide/) instead mock code duplication

File: src/test/resources/pre/sql/projects.sql [hunk 2]
```diff
@@ -36,5 +36,5 @@ INSERT INTO projects.secret(project,
                             value)
 VALUES ('74bb5ec8-0e6b-4618-bfa4-a0b76b7b312d',
         'key',
-        'value')
+        'Iohe7UjU5MjHcxqsKk9vMLTlYHFCXHCLl4q0EjfBbfzmiYtXI1Vfw3KRUQEmbeVh')
```
Comment (h1alexbel) [inline src/test/resources/pre/sql/projects.sql:39]: lets document our `symetric` encryption approach we using here

**Review by h1alexbel — CHANGES_REQUESTED** [review]
@hizmailovich take a look, please

File: src/test/resources/pre/sql/projects.sql [hunk 3]
```diff
@@ -31,6 +31,7 @@ VALUES ('04986038-6e38-4928-b12e-644c99f9cadc',
         'OPENED')
 ON CONFLICT (id) DO NOTHING;
 
+-- value encrypted using algorithm PBEWithHMACSHA512AndAES_256
```
Comment (h1alexbel) [inline src/test/resources/pre/sql/projects.sql:34]: to document I mean to add useful information to our `README.md`

**Review by h1alexbel — CHANGES_REQUESTED** [review]
@hizmailovich take a look, please

**Comment by h1alexbel** [comment]
@hizmailovich please update your token

**Review by h1alexbel — APPROVED** [review]

**Comment by h1alexbel** [comment]
@rultor merge

**Comment by hizmailovich** [comment]
@rultor merge

**Pull request merged**